### PR TITLE
Add support for Decimal number.

### PR DESCRIPTION
**Pull request by xaviersolau**

### CODE REVIEW

File: Verse/src/Schemas/JSON/DecoderConverter.cs [hunk 1]
```diff
@@ -28,15 +28,15 @@ internal class DecoderConverter : IDecoderConverter<JSONValue>
...
+			if (!this.converters.TryGetValue(typeof (TTo), out var box))
 				throw new InvalidCastException(string.Format(CultureInfo.InvariantCulture, "cannot convert JSON value into '{0}', please register a converter using schema's SetDecoderConverter method", typeof (TTo)));
 
 			return (Converter<JSONValue, TTo>)box;
 		}
 
 		public void Set<TTo>(Converter<JSONValue, TTo> converter)
 		{
-            this.converters[typeof (TTo)] = converter ?? throw new ArgumentNullException(nameof(converter));
+			this.converters[typeof (TTo)] = converter ?? throw new ArgumentNullException(nameof(converter));
```
Comment (r3c) [inline Verse/src/Schemas/JSON/DecoderConverter.cs]: [NIT] You have a few unwanted space/tab replacement in the review, could you please try not to change spacing?

File: Verse/src/Schemas/JSON/JSONValue.cs [hunk 2]
```diff
@@ -16,7 +15,12 @@ namespace Verse.Schemas.JSON
 		/// <summary>
 		/// Numeric value (only set if type is number).
 		/// </summary>
-		public readonly double Number;
+		public double Number => this.numberMantissa * this.number10PowExponent;
+
+		public decimal NumberAsDecimal => this.numberMantissa * (decimal) this.number10PowExponent;
```
Comment (r3c) [inline Verse/src/Schemas/JSON/JSONValue.cs]: As this class is public, could you please insert a comment to explain what's the benefit of this property?

File: Verse/src/Schemas/JSON/JSONValue.cs [hunk 3]
```diff
@@ -40,17 +44,63 @@ namespace Verse.Schemas.JSON
...
+				var mantissa = (long) (number / number10PowExponent);
+
+				var res = new JSONValue(JSONType.Number, default, mantissa, number10PowExponent, default);
+				return res;
+			}
+		}
+
+		private static int ComputeExponent(double value)
+		{
+			// Extract the exponent from the double:
```
Comment (r3c) [inline Verse/src/Schemas/JSON/JSONValue.cs]: Thank you for the comments :)

File: Verse/src/Schemas/JSON/DecoderConverter.cs (hunk 1 above)
Comment (xaviersolau) [reply Verse/src/Schemas/JSON/DecoderConverter.cs]: removed spacing changes

File: Verse/src/Schemas/JSON/JSONValue.cs (hunk 2 above)
Comment (xaviersolau) [reply Verse/src/Schemas/JSON/JSONValue.cs]: done

**Pull request merged**